File: Python Pattern Programs/Wave Patterns/Pattern 13.py
wH = 4  # 1
wL = 4  # 2
pc = wH - 1  # 3

for x in range(0, wH + 1):
    pchar = ord('a')  # 4
    for y in range(0, (wH * wL * 2)):
        if (y % (wH * 2) == pc or
                y % (wH * 2) == wH + x):
            print(chr(pchar), end="")
        else:
            print(" ", end="")  # single space
        pchar += 1  # 5
        if (pchar > ord('z')):  # 6
            pchar = pchar - 26  # 7
    pc -= 1
    print()

    """
1) change value to increase/decrease the height of the wave
2) change value to increase/decrease the length of the wave
3) cond for printing
4) set printing character
5) increment print char
6) reset print char to 'a'
    """

File: JavaPatternPrograms/WavePatterns/Pattern 13.java
class Wave_13
{
   public static void main(String args[])
   {
    int wH = 4; //wave height-> change value to increase or decrease the height of wave


    int wL = 4; //wave length->change value to increase or decrease the length of wave


    int x = wH - 1; //if cond for printing

    int cp; //print char

    for (int i = 0;i <= wH;i++)
    {
     cp = 'a'; // set print char.

     for (int j = 0;j < wH * wL * 2;j++)
     {

        if (j % (wH * 2) == x || j % (wH * 2) == wH + i)
        {
        System.out.print((char)cp);
        }
        else
        {
         System.out.print(" ");
        }

      cp++; // increment print char

      /reset print char to 'a'/
      if (cp > 'z')
      {
         cp = cp - 26;
      }

     }
       x--;
     System.out.println();
    }
   
   }
}